app/models/annotation.rb

```diff
@@ -40,7 +40,12 @@ def get_data(include_creator=false)
         annotation_text.annotation_category&.annotation_category_name,
       type: self.class.name,
       number: annotation_number,
-      is_remark: is_remark
+      is_remark: is_remark,
+      deduction: annotation_text.deduction.nil? ? 'undefined' : annotation_text.deduction,
+      criterion_name: annotation_text&.annotation_category&.flexible_criterion.nil? ? 'undefined' :
```

Comment:
The first `&.` should just be a `.`, since the `self.annotation_text` association should never be `nil`.